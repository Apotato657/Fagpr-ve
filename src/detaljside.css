.detalj-wrapper {
    margin: 2rem 0;
}