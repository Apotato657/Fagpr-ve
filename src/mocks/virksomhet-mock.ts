import {Virksomhet} from "../model/virksomhet";

export const MOCK_ENHET: Virksomhet = {
    organisasjonsnummer: '987654321',
    navn: 'MOZFELDT REGNSKAP AS',
    organisasjonsform: {
        kode: 'Enkeltpersonforetak',
        beskrivelse: 'JA?',
    },
    forretningsadresse: {
        land: 'Norge',
        landkode: 'NO',
        postnummer: '3041',
        poststed: 'DRAMEN',
        adresse: [
            ''
        ],
        kommune: 'DRAMEN',
        kommunenummer: '3301'
    },
    antallAnsatte: 18,
    konkurs: false,
    registreringsdatoEnhetsregisteret: '2021-3-16',
    postadresse: {
        land: 'Norge',
        landkode: 'NO',
        postnummer: '3041',
        poststed: 'DRAMEN',
        adresse: [
            ''
        ],
        kommune: 'DRAMEN',
        kommunenummer: '3301'
    }
}
