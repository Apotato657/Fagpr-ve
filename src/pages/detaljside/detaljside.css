.detalj-wrapper {
    margin: 2rem 0;
}

.innlogging {
    margin-top: 1rem;
    display: flex;
    justify-content: flex-end;
}

.innlogging div {
    display: flex;
    align-items: center;
}