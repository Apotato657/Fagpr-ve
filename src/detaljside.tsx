import {Card, Heading, Paragraph} from "@digdir/designsystemet-react";
import {DetailGroup, DetailRow} from "./components/detail-row";
import './detaljside.css'

export function Detaljside() {

    return (
    <>
        <Heading level={1} data-size={"xl"}>Navn p√• virksomhet</Heading>
        <Paragraph>Orgnummer</Paragraph>

        <Card className='detalj-wrapper'>
            <Card.Block>
                <Heading level={2} data-size={"md"}>Opplysninger om virksomheten</Heading>
            </Card.Block>
            <Card.Block>
                <DetailGroup>
                    <DetailRow label='Foretningsadresse:'>

                    </DetailRow>
                    <DetailRow label='Kommune, land:'>

                    </DetailRow>
                    <DetailRow label='Organisasjonsform:'>

                    </DetailRow>
                    <DetailRow label='under konkursbehandling:'>

                    </DetailRow>
                    <DetailRow label='Antall ansatte:'>

                    </DetailRow>
                    <DetailRow label='Aktivitet:'>

                    </DetailRow>
                    <DetailRow label='Registreringsdato i Enhetsregisteret:'>

                    </DetailRow>
                    <DetailRow label='Roller:'>

                    </DetailRow>
                </DetailGroup>

            </Card.Block>
        </Card>
    </>
    )
}