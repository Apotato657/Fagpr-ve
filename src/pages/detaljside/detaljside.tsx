import {Button, Card, Heading, Link, Paragraph} from "@digdir/designsystemet-react";
import {DetailGroup, DetailRow} from "../../components/detail-row";
import './detaljside.css'
import React, {useEffect, useState} from "react";
import {useAuth0} from "@auth0/auth0-react";
import LogoutButton from "../../Auth0/button-utlogging";
import LoginButton from "../../Auth0/button-innlogging";
import {Virksomhet} from "../../model/virksomhet";
import {fetchSingelVirksomhet} from "../../virksomhet/fetch-virksomhet";
import {useParams} from "react-router";
import {StringUtils} from "../../utils/string-utils";

export function Detaljside() {

    const {orgnummer} = useParams<{orgnummer: string}>();
    const { user, isAuthenticated} = useAuth0();
    const [virksomhet, setVirksomhet] = useState<Virksomhet>()
    const [error, setError] = useState<string>()

    useEffect(() => {
        (async () => {
            const response = await fetchSingelVirksomhet(orgnummer);
            if (response.status === "success") {
                setVirksomhet(response.virksomhet as Virksomhet);
            } else {
                setError(response.error);
                setVirksomhet(undefined);
            }
        })();
    }, [orgnummer]);

    return (
        <>
            <header className='innlogging'>
                {isAuthenticated ? (
                    <div>
                        <Paragraph>Innlogget: {user?.name}</Paragraph>
                        <LogoutButton/>
                    </div>
                ) : (
                    <LoginButton/>
                )
                }
            </header>
            <Heading level={1} data-size={"xl"}>{virksomhet?.navn}</Heading>
            <Heading level={3}>{StringUtils.formatOrgNummer(virksomhet?.organisasjonsnummer ?? '')}</Heading>

            <Card className='detalj-wrapper'>
                <Card.Block>
                    <Heading level={2} data-size={"md"}>Opplysninger om virksomheten</Heading>
                    {error &&
                        <Heading level={2}>{error}</Heading>
                    }
                </Card.Block>
                <Card.Block>
                    <DetailGroup>
                        <DetailRow label={virksomhet?.forretningsadresse?.adresse ? 'Forretningsadresse' : 'Postadresse'}>
                            {virksomhet?.forretningsadresse.adresse ? virksomhet?.forretningsadresse?.adresse : virksomhet?.postadresse.adresse}<br/> {virksomhet?.forretningsadresse.kommunenummer} {virksomhet?.forretningsadresse.kommune}
                        </DetailRow>
                        <DetailRow label='Organisasjonsform:'>
                            {virksomhet?.organisasjonsform.beskrivelse}
                        </DetailRow>
                        {virksomhet?.konkurs &&
                            <DetailRow label='Under konkursbehandling:'>
                                {virksomhet?.konkurs}
                            </DetailRow>
                        }
                            <DetailRow label='Antall ansatte:'>
                                {virksomhet?.antallAnsatte !== undefined
                                ? virksomhet.antallAnsatte : 'Ingen registrerte ansatte'}
                            </DetailRow>

                        <DetailRow label='Aktivitet:'>
                            {virksomhet?.aktivitet}
                        </DetailRow>
                        <DetailRow label='Registreringsdato i Enhetsregisteret:'>
                            {virksomhet?.registreringsdatoEnhetsregisteret}
                        </DetailRow>
                        <DetailRow label='Roller:'>

                        </DetailRow>
                    </DetailGroup>

                </Card.Block>
            </Card>

            <Button className={'buttonWrapper'} asChild>
                <Link href={'/'}>
                    Tilbake til s√∏keside
                </Link>
            </Button>
        </>
    )
}