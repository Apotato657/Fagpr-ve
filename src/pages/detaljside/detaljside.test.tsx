import {render} from "@testing-library/react";
import {Detaljside} from "./detaljside";
import fetchMock from "jest-fetch-mock";
import {MOCK_ENHET} from "../../mocks/virksomhet-mock";

describe('test detaljside', () => {
    const setup = () => {
        const view = render(<Detaljside/>)
        return {view};
    }

    it('should render correctly', () => {
        const mockVirksomhetApiRespons = {virksomheter: MOCK_ENHET};
        fetchMock.mockResponseOnce(JSON.stringify(mockVirksomhetApiRespons));


        const view = setup();
        expect(view).toMatchSnapshot();
    });
})