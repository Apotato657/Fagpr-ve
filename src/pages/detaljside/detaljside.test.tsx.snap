// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test detaljside should render correctly 1`] = `
{
  "view": {
    "asFragment": [Function],
    "baseElement": <body>
      <div>
        <header
          class="innlogging"
        >
          <button
            class="ds-button"
            data-variant="primary"
            type="button"
          >
            Log In
          </button>
        </header>
        <h1
          class="ds-heading"
          data-size="xl"
        />
        <h3
          class="ds-heading"
        />
        <div
          class="ds-card detalj-wrapper"
          data-variant="default"
        >
          <div
            class="ds-card__block"
          >
            <h2
              class="ds-heading"
              data-size="md"
            >
              Opplysninger om virksomheten
            </h2>
          </div>
          <div
            class="ds-card__block"
          >
            <dl>
              <dt>
                Postadresse
              </dt>
              <dd>
                <br />
                 
                 
              </dd>
              <dt>
                Organisasjonsform:
              </dt>
              <dd />
              <dt>
                Antall ansatte:
              </dt>
              <dd>
                Ingen registrerte ansatte
              </dd>
              <dt>
                Aktivitet:
              </dt>
              <dd />
              <dt>
                Registreringsdato i Enhetsregisteret:
              </dt>
              <dd />
              <dt>
                Roller:
              </dt>
              <dd />
            </dl>
          </div>
        </div>
        <a
          class="ds-link ds-button buttonWrapper"
          data-variant="primary"
          href="/"
        >
          Tilbake til søkeside
        </a>
      </div>
    </body>,
    "container": <div>
      <header
        class="innlogging"
      >
        <button
          class="ds-button"
          data-variant="primary"
          type="button"
        >
          Log In
        </button>
      </header>
      <h1
        class="ds-heading"
        data-size="xl"
      />
      <h3
        class="ds-heading"
      />
      <div
        class="ds-card detalj-wrapper"
        data-variant="default"
      >
        <div
          class="ds-card__block"
        >
          <h2
            class="ds-heading"
            data-size="md"
          >
            Opplysninger om virksomheten
          </h2>
        </div>
        <div
          class="ds-card__block"
        >
          <dl>
            <dt>
              Postadresse
            </dt>
            <dd>
              <br />
               
               
            </dd>
            <dt>
              Organisasjonsform:
            </dt>
            <dd />
            <dt>
              Antall ansatte:
            </dt>
            <dd>
              Ingen registrerte ansatte
            </dd>
            <dt>
              Aktivitet:
            </dt>
            <dd />
            <dt>
              Registreringsdato i Enhetsregisteret:
            </dt>
            <dd />
            <dt>
              Roller:
            </dt>
            <dd />
          </dl>
        </div>
      </div>
      <a
        class="ds-link ds-button buttonWrapper"
        data-variant="primary"
        href="/"
      >
        Tilbake til søkeside
      </a>
    </div>,
    "debug": [Function],
    "findAllByAltText": [Function],
    "findAllByDisplayValue": [Function],
    "findAllByLabelText": [Function],
    "findAllByPlaceholderText": [Function],
    "findAllByRole": [Function],
    "findAllByTestId": [Function],
    "findAllByText": [Function],
    "findAllByTitle": [Function],
    "findByAltText": [Function],
    "findByDisplayValue": [Function],
    "findByLabelText": [Function],
    "findByPlaceholderText": [Function],
    "findByRole": [Function],
    "findByTestId": [Function],
    "findByText": [Function],
    "findByTitle": [Function],
    "getAllByAltText": [Function],
    "getAllByDisplayValue": [Function],
    "getAllByLabelText": [Function],
    "getAllByPlaceholderText": [Function],
    "getAllByRole": [Function],
    "getAllByTestId": [Function],
    "getAllByText": [Function],
    "getAllByTitle": [Function],
    "getByAltText": [Function],
    "getByDisplayValue": [Function],
    "getByLabelText": [Function],
    "getByPlaceholderText": [Function],
    "getByRole": [Function],
    "getByTestId": [Function],
    "getByText": [Function],
    "getByTitle": [Function],
    "queryAllByAltText": [Function],
    "queryAllByDisplayValue": [Function],
    "queryAllByLabelText": [Function],
    "queryAllByPlaceholderText": [Function],
    "queryAllByRole": [Function],
    "queryAllByTestId": [Function],
    "queryAllByText": [Function],
    "queryAllByTitle": [Function],
    "queryByAltText": [Function],
    "queryByDisplayValue": [Function],
    "queryByLabelText": [Function],
    "queryByPlaceholderText": [Function],
    "queryByRole": [Function],
    "queryByTestId": [Function],
    "queryByText": [Function],
    "queryByTitle": [Function],
    "rerender": [Function],
    "unmount": [Function],
  },
}
`;
